tests/api/test_shop.py

```diff
@@ -830,3 +830,154 @@ def test_mutation_update_company_address_remove_address_without_address(
 
     site_settings.refresh_from_db()
     assert not site_settings.company_address
+
+
+def test_staff_notification_query(
+    staff_api_client,
+    staff_user,
+    permission_manage_settings,
+    staff_notification_recipient,
+):
+    query = """
+        {
+            shop {
+                staffNotificationRecipient {
+                    active
+                    email
+                    user {
+                        firstName
+                        lastName
+                        email
+                    }
+                }
+            }
+        }
+    """
+
+    response = staff_api_client.post_graphql(
+        query, permissions=[permission_manage_settings]
+    )
+    content = get_graphql_content(response)
+
+    assert content["data"]["shop"]["staffNotificationRecipient"] == [
+        {
+            "active": True,
+            "email": staff_user.email,
+            "user": {
+                "firstName": staff_user.first_name,
+                "lastName": staff_user.last_name,
+                "email": staff_user.email,
+            },
+        }
+    ]
+
+
+def test_staff_notification_create_mutation(
+    staff_api_client, staff_user, permission_manage_settings
+):
+    query = """
+        mutation StaffNotificationRecipient ($user: ID!) {
+            staffNotificationRecipientCreate(input: {user: $user}) {
+                staffNotificationRecipient {
+                    active
+                    email
+                    user {
+                        firstName
+                        lastName
+                        email
+                    }
+                }
+            }
+        }
+    """
+    user_id = graphene.Node.to_global_id("User", staff_user.id)
+    variables = {"user": user_id}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_settings]
+    )
+    content = get_graphql_content(response)
+
+    assert content["data"]["staffNotificationRecipientCreate"] == {
+        "staffNotificationRecipient": {
+            "active": True,
+            "email": staff_user.email,
+            "user": {
+                "firstName": staff_user.first_name,
+                "lastName": staff_user.last_name,
+                "email": staff_user.email,
+            },
+        }
+    }
+
+
+def test_staff_notification_create_mutation_with_staffs_email(
+    staff_api_client, staff_user, permission_manage_settings
+):
+    query = """
+        mutation StaffNotificationRecipient ($staff_email: String!) {
+            staffNotificationRecipientCreate(input: {staffEmail: $staff_email}) {
+                staffNotificationRecipient {
+                    active
+                    email
+                    user {
+                        id
+                        firstName
+                        lastName
+                        email
+                    }
+                }
+            }
+        }
```

Comment:
It would be better to extract the mutation query outside of the test and have `staffEmail` + `email` optional.

Reply:
done